### PR TITLE
fix(ds): Fix packet id -> sequence number translation

### PR DESCRIPTION
**Pull request by ieQu1**
Fixes <issue-or-jira-number>

## Summary
copilot:summary

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx/src/emqx_persistent_message_ds_replayer.erl [hunk 1]
```diff
@@ -215,3 +233,55 @@ shuffle(L0) ->
...
+%% This test only tests boundary conditions (to make sure property-based test didn't skip them):
+packet_id_to_seqno_test() ->
+    %% Packet ID = 1; first epoch:
+    ?assertEqual(1, packet_id_to_seqno(1, 1)),
+    ?assertEqual(1, packet_id_to_seqno(10, 1)),
+    ?assertEqual(1, packet_id_to_seqno(1 bsl 16 - 1, 1)),
+    %% Packet ID = 1; second and 3rd epochs:
+    ?assertEqual(1 bsl 16 + 1, packet_id_to_seqno(1 bsl 16 + 1, 1)),
+    ?assertEqual(2 bsl 16 + 1, packet_id_to_seqno(2 bsl 16 + 1, 1)),
+    %% Packet ID = 16#ffff:
```
Comment (thalesmg) [inline apps/emqx/src/emqx_persistent_message_ds_replayer.erl:248]: Nit: to assert the behavior at the boundaries:

```suggestion
    ?assertEqual(1, packet_id_to_seqno(1 bsl 16 - 1, 1)),
    ?assertEqual(1, packet_id_to_seqno(1 bsl 16, 1)),
    %% Packet ID = 1; second and 3rd epochs:
    ?assertEqual(1 bsl 16 + 1, packet_id_to_seqno(1 bsl 16 + 1, 1)),
    ?assertEqual(1 bsl 16 + 1, packet_id_to_seqno(2 bsl 16, 1)),
    ?assertEqual(2 bsl 16 + 1, packet_id_to_seqno(2 bsl 16 + 1, 1)),
    %% Packet ID = 16#ffff:
```

File: apps/emqx/src/emqx_persistent_message_ds_replayer.erl [hunk 2]
```diff
@@ -215,3 +233,55 @@ shuffle(L0) ->
...
+
+next_seqno_gen() ->
+    ?LET(
+        {Epoch, Offset},
+        {non_neg_integer(), non_neg_integer()},
+        Epoch bsl 16 + Offset
+    ).
+
+seqno_gen(NextSeqNo) ->
+    WindowSize = 1 bsl 16 - 2,
```
Comment (thalesmg) [inline apps/emqx/src/emqx_persistent_message_ds_replayer.erl:282]: Q: isn't the maximum window size / [receive maximum](https://github.com/emqx/emqx/pull/11720#discussion_r1386127718) `0xffff` / `1 bsl 16 - 1`?

**Review by thalesmg — APPROVED** [review]

**Review by thalesmg — DISMISSED** [review]

**Review by thalesmg — APPROVED** [review]
